src/Dfe.PlanTech.AzureFunctions/Mappings/SectionMapper.cs

```diff
@@ -25,4 +28,30 @@ public class SectionMapper(EntityRetriever retriever, EntityUpdater updater, Cms
         return values;
     }
 
+
+    public override async Task<MappedEntity> MapEntity(CmsWebHookPayload payload, CmsEvent cmsEvent, CancellationToken cancellationToken)
+    {
+        var mappedEntity = await base.MapEntity(payload, cmsEvent, cancellationToken);
+
+        if (!mappedEntity.AlreadyExistsInDatabase)
+        {
+            return mappedEntity;
+        }
+
+        return RemoveSectionFromRemovedQuestions(mappedEntity);
+    }
+
+    private static MappedEntity RemoveSectionFromRemovedQuestions(MappedEntity mappedEntity)
+    {
+        var (incoming, existing) = mappedEntity.GetTypedEntities<SectionDbEntity>();
+
+        var questionsToRemove = existing.Questions.Where(existingQuestion => !incoming.Questions.Exists(incomingQuestion => incomingQuestion.Id == existingQuestion.Id));
+
+        foreach (var answerToRemove in questionsToRemove)
```

Comment:
Can we rename this `questionToRemove`?
```suggestion
        foreach (var questionToRemove in questionsToRemove)
```

Reply:
Good spot. Sorted.